interfaces/builtin/desktop_launch.go

```diff
@@ -42,6 +42,9 @@ const desktopLaunchConnectedPlugAppArmor = `
 /var/lib/snapd/desktop/applications/{,*} r,
 /var/lib/snapd/desktop/icons/{,**} r,
 
+# Allow access to all snap metadata
+/snap/*/*/** r,
```

Comment:
This is not going to help on systems where snaps are mounted in `/var/lib/snapd/snap`. Ultimately I think that's okay as desktop-launch is primarily useful on Ubuntu Core systems.

It won't grant access to the `/snap/${SNAP_NAME}/current` symlinks but I think that is also ultimately okay, since the `Icon=` lines in the sanitised desktop files will be rewritten to the actual mount dir containing the revision number.